propagator/xray/lib/opentelemetry/propagator/xray.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# OpenTelemetry is an open source observability framework, providing a
+# general-purpose API, SDK, and related tools required for the instrumentation
+# of cloud-native software, frameworks, and libraries.
+#
+# The OpenTelemetry module provides global accessors for telemetry objects.
+# See the documentation for the `opentelemetry-api` gem for details.
+
+require_relative './xray/text_map_extractor'
+require_relative './xray/text_map_injector'
+
+module OpenTelemetry
+  # Namespace for OpenTelemetry propagator extension libraries
+  module Propagator
+    # Namespace for OpenTelemetry XRay propagation
+    module XRay
+      extend self
+
+      DEBUG_CONTEXT_KEY = Context.create_key('xray-debug-key')
+      private_constant :DEBUG_CONTEXT_KEY
+
+      # @api private
+      # Returns a new context with the xray debug flag enabled
+      def context_with_debug(context)
+        context.set_value(DEBUG_CONTEXT_KEY, true)
+      end
+
+      # @api private
+      # Read the XRay debug flag from the provided context
+      def debug?(context)
+        context.value(DEBUG_CONTEXT_KEY)
+      end
+
+      # @api private
+      # Convert an id from a hex encoded string to byte array, optionally left
+      # padding to the correct length. Assumes the input id has already been
+      # validated to be 16 or 32 characters in length.
+      def to_trace_id(hex_id)
+        Array(hex_id[2..8] + hex_id[10..hex_id.length - 1]).pack('H*')
```

Comment:
```suggestion
        Array(hex_id[2..8] + hex_id[10..]).pack('H*')
```

Reply:
This syntax seems to have been introduced in Ruby 2.6? Is support for 2.5 being dropped?